qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -102,7 +129,13 @@ def enable_account(
                            'account.')
             return None
 
-        return providers[0]
+        default_provider = providers[0]
+
+        # Only filter if at least one is set.
+        if hub or group or project:
+            default_provider = self.get_provider(hub=hub, group=group, project=project)
```

Comment:
You should check if the provider for the h/g/p can be found and raise an error if not, so the user doesn't accidentally use the wrong provider. For example, a user makes a typo when specifying h/g/p and doesn't find out until the job finally finishes 7 hours later lol.